account_banking_pain_base/models/banking_export_pain.py

```diff
@@ -289,16 +289,16 @@ def generate_party_agent(
         """Generate the piece of the XML file corresponding to BIC
         This code is mutualized between TRF and DD"""
         assert order in ('B', 'C'), "Order can be 'B' or 'C'"
-        try:
-            bic = self._prepare_field(
-                '%s BIC' % party_type_label, bic, eval_ctx, gen_args=gen_args)
+        bic = self._prepare_field(
+            '%s BIC' % party_type_label, bic, eval_ctx, gen_args=gen_args)
+        if bic:
```

Comment:
I think it's the `bic` argument that you need to check instead of the result of this method

Reply:
Ah yes! Thanks

Reply:
Well, the contents of the bic argument is something like `'line.bank_id.bank.bic or line.bank_id.bank_bic'` when evaluating the customers' BIC code. 
I don't really see how/what we should check. I think it's better to comment out the `if order == 'C':` part because we forcibly require the BIC code when the bank account is not in EU or EFTA country group lists.

Reply:
Also, there is a comment right below the last if here:
```
# for Credit Transfers, in the 'C' block, if BIC is not provided,
# we should not put the 'Creditor Agent' block at all,
# as per the guidelines of the EPC
```
I believe their intent here is to do the same (Although not completely correct since the BIC is still required for transactions outside of the EU or EFTA.